### PR TITLE
Fix combining of `--editable` and `--force` flag

### PR DESCRIPTION
**Pull request by potiuk**
* [x] I have added an entry to `docs/changelog.md`

## Summary of changes

The 1.3.0 version of `pipx` (with #933) introduced a but where the `--force-reinstall` flag caused failure when also the `--editable` flag was used - because it was adding the `--force-reinstall` flag after `--editable` one (and `--editable` flag expects url/path to the Python package to install to follow it.

The change fixes it by adding the `--force-reinstall` flag at the beginning rather than at the end of arguments to avoid this kind of problem also in case other flags might have similar problem.

Fixes: #1122

## Test plan

Tested by running
```
pytest tests/test_install.py::test_force_install_changes_editable
```

Screenshot 2023-12-02 at 20 11 27

### CODE REVIEW

File: tests/test_install.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 TEST_DATA_PATH = "./testdata/test_package_specifier"
 
+SOURCE_DIR = Path(__file__).parent.parent.resolve()
```
Comment (gaborbernat) [inline tests/test_install.py:16]: Remove this and used the `root` fixture instead.

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: tests/test_install.py (hunk 1 above)
Comment (potiuk) [reply tests/test_install.py:16]: Ah nice. TIL

Comment (potiuk) [reply tests/test_install.py]: Fixed.

**Comment by gaborbernat** [comment]
No yanking but https://github.com/pypa/pipx/actions/runs/7072334702

**Comment by potiuk** [comment]
👍 🎉 🎉 🎉 🎉 🎉 👍

**Comment by potiuk** [comment]
And released ! :)

**Pull request merged**